### ISSUE TITLE
Support for CrossValidation: Enhancement Request

### ISSUE THREAD

**Issue by Alan-Penkar** [ray-project/lightgbm_ray#42]
I am using RayDP with Spark and am using this package with Ray Tune for HyperParameter Optimization with the lightGBM regressor.  Unless there is something I'm missing, there's no way to use lgbm's native cross validation as in Ray's [examples](https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/lightgbm_example.py), this would be a huge help to model accuracy when training large models.

**Comment by Yard1** [comment]
I am not sure whether there's a built-in way of doing that in other distributed lightgbm implementations. I think the `cv` function calls `train` underneath - as a workaround, you should be able to replace that with lightgbm-ray's `train` function.

**Comment by Alan-Penkar** [comment]
Thanks for the quick response - I don't actually see lightgbm.train being called in the .cv function, but I will continue looking through the nested calls.  The bigger question is how to make the `_make_n_folds` function within lightgbm.cv compatible with the RayDMatrix objects that are used by lightgbm_ray?  I haven't seen any methods on a RayDMatrix that would lead me to believe data manipulation would be straightforward, but if I've missed it I'd certainly appreciate you pointing me in the right direction.
